rails_model/migrations.md

```diff
@@ -14,40 +14,79 @@
 - migrations
 
 ## Useful Commands
+- $ rails generate migration file_name_here
 - $ rails db:migrate
 
 ## Additional Resources
-- <a href="https://edgeguides.rubyonrails.org/active_record_migrations.html" target="blank">Rails documentation on Migrations</a>
-- <a href="https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method" target="blank">Change Method</a>
+- <a href="https://edgeguides.rubyonrails.org/active_record_migrations.html" target="blank">Official Rails documentation on Migrations</a>
+- <a href="https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method" target="blank">A list of all the change definitions</a>
+
+## What is a migration file?
+A migration is a file that is generated through a rails command that will allow you to change the schema. This can include (but not limited to) editing a column's name or data type, adding a new column or removing a column. 
+
+When working with projects that utilize databases, your required columns or datatypes may need to change to accomodate new specifications which requires changing the shape of your schema. 
+
+Directly editing the schema file is considered bad practice as the file itself is not designed to be edited and is just a representation of the current state of your schema.
+
+## How do we edit the schema with a migration file?
+
+We are able to accomplish edits to the schema by running a rails migration generate command using proper naming conventions that communicate the intent of your changes. 
+
+## Naming Conventions
+- Migration names are PascalCased or snake_cased
+- Migration names start with an action that describes the purpose of the migration (i.e. create, add, remove, etc)
+- The migration name ends with the name of the table and should be plural (add_name_to_users)
+
+
+An example of this using snake_case: 
+```
+$ rails generate migration add_foreign_key_to_products
+```
+
+Alternatively, we can use PascalCase:
+
+```
+$ rails generate migration AddForeignKeytoProducts
+```
+
+Either way is fine, but it is important to make a decision to only use one or the other.
+
+Running this command in your terminal will generate a new file to your migration directory. Editing this file and adding the changes you want within the "def change" method will allow you to change the schema as needed! 
+
+## Set up from scratch
+
+In the exercise below, we'll be creating a brand new rails application, but we will do it a bit differently. In the model we create we will leave out all of the columns that we would normally include in the beginning thus requiring us to create a migration file to add them in. 
 
-## Set Up
 
 #### Creating a new Rails app:
+
```

Comment:
This line needs to match your app name - and can I suggest making the app name something like migrations_practice or something more descriptive?